x/stakeibc/keeper/host_zone.go

```diff
@@ -313,6 +318,42 @@ func (k Keeper) RemoveValidatorFromHostZone(ctx sdk.Context, chainId string, val
 	return errorsmod.Wrapf(types.ErrValidatorNotFound, errMsg)
 }
 
+// Checks if any validator's portion of the weight is greater than the cap
+func (k Keeper) CheckValidatorWeightsBelowCap(ctx sdk.Context, validators []*types.Validator) error {
+	// If there's only a few validators, don't enforce this yet
+	if len(validators) < MinValidatorsBeforeWeightCapCheck {
+		return nil
+	}
+
+	params := k.GetParams(ctx)
+	validatorWeightCap := float64(params.ValidatorWeightCap)
+
+	// Store a map of each validator weight, as well as the total
+	totalWeight := float64(0)
+	weightsByValidator := map[string]float64{}
+	for _, validator := range validators {
+		weightsByValidator[validator.Address] = float64(validator.Weight)
+		totalWeight += float64(validator.Weight)
+	}
+
+	// If the total validator weights are 0, exit prematurely
+	if totalWeight == 0 {
+		return nil
+	}
+
+	// Check if any validator exceeds the cap
+	for _, address := range utils.StringMapKeys[float64](weightsByValidator) {
```

Comment:
this was because of the non-determinism issue with maps in go!

I actually don't think this would re-create the bug cause in the past I think we've only seen it if it could lead to different state transitions.

But considering this throws an error and that error could be different depending on how it loops (imagine a scenario where multiple validators exceed the cap), it felt _slightly_ safe to do the sorting first

Reply:
Yeah that makes sense, let's do it this way!